Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAuditdPlugin.py

```diff
@@ -49,94 +68,299 @@ def Get_Marshall(Ensure):
-    if state == None:
+    state = GetEnsure(WorkspaceId)
+    if state is None:
         return (-1, "Absent")
-        
-    if state and txt == desired_txt:
-        return (0, "Present")
     else:
-        return (0, "Absent")
+        return (0, state)
+
+def GetOtherOutconfCounts(WorkspaceId):
+    tcount = 0
+    wcount = 0
+    for name in os.listdir(AUOMS_OUTCONF_DIR):
+        if name.endswith(".conf"):
+            if name != WorkspaceId + ".conf":
+                tcount += 1
+                if IsValidWorkspaceId(name[:-5]):
+                    wcount += 1
+    return (tcount, wcount)
+
+def GetDesiredAuditRules(AuditVersion):
+    if AuditVersion == 1:
+        desired_rules = ReadFile(RESOURCE_AUDIT_RULES_V1)
+    else:
+        desired_rules = ReadFile(RESOURCE_AUDIT_RULES_V2)
+
+    if platform.machine() != "x86_64" and desired_rules != None:
+        lines = desired_rules.splitlines()
+        new_lines = []
+        for line in lines:
+            if not 'b64' in line:
+                new_lines.append(line)
+        desired_rules = "\n".join(lines) + "\n"
+    return desired_rules
+
+def GetDesiredState(WorkspaceId, Ensure, AuditVersion, CurrentPluginState):
+    desired_plugin_state = "no"
+    desired_rules = None
+    desired_outconf = None
+    desired_plugin_conf = None
+
+    (other_conf_count, other_ws_conf_count) = GetOtherOutconfCounts(WorkspaceId)
+
+    if Ensure == "Present":
+        desired_plugin_state = "yes"
+        desired_rules = GetDesiredAuditRules(AuditVersion)
+        desired_outconf = ReadFile(RESOURCE_AUOMS_CONF)
+        desired_plugin_conf = ReadFile(RESOURCE_OMSAGENT_CONF)
+
+        if desired_rules is None or desired_outconf is None or desired_plugin_conf is None:
+            # Failed to determine state
+            return (None, None, None, None)
+
+        desired_outconf = desired_outconf.replace(WORKSPACE_ID_TMPL_STR, WorkspaceId)
+        desired_plugin_conf = desired_plugin_conf.replace(WORKSPACE_ID_TMPL_STR, WorkspaceId)
+    else:
+        if other_conf_count > 0:
+            if other_ws_conf_count > 0:
+                # There are other workspace outconfs present, so the desired state is "yes"
+                desired_plugin_state = "yes"
+            else:
+                # If the only other outconfs are non-OMS, then we don't
+                # want to change the auoms plugin state
+                desired_plugin_state = CurrentPluginState
+        # If there are any other workspaces with auoms outconf files, then
+        # we don't want to remove the audit rules
+        if other_ws_conf_count > 0:
+            desired_rules = GetDesiredAuditRules(AuditVersion)
+
+    return (desired_plugin_state, desired_rules, desired_outconf, desired_plugin_conf)
+
+def GetEnsure(WorkspaceId):
+    (audit_version, plugin_state,
+     actual_audit_rules, actual_outconf, actual_plugin_conf) = GetState(WorkspaceId)
+    if audit_version is None:
+        return None
+
+    Ensure = "Absent"
```

Comment:
What is the point of this line if you set it to "Present" no matter what on line 272?